### ISSUE TITLE
Loss is computed as "nan" with VGG16 for Fast/Faster R-CNN

### ISSUE THREAD

**Issue by kyoro1** [Microsoft/CNTK#2456]
In case of using `VGG16` as base_model for Fast/Faster R-CNN with CNTK2.2, loss is not computed correctly as follows, although I've not encountered this issue with AlexNet. What the difference is just the base model.

-Environment:
  - `CNTK`: 2.2
  - Processing: `GPU`(NC6 in Azure)
- Result
  - base_model: 
    - AlexNet -> OK, 
    - VGG16 -> appeared `nan`, No images was computed correctly after completing the whole iterations.

Then, can we share the appropriate setting in case of `VGG16`?

```sh
C:\git\ver2.2\CNTK\Examples\Image\Detection\FastRCNN (master)
(cntk2.2) λ python run_fast_rcnn.py
Selected GPU[0] Tesla K80 as the process wide default device.
Using base model:   VGG16
lr_per_sample:      [0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.001, 0.001, 0.001, 0.001, 0.001, 0.001, 0.001, 0.001, 0.001, 0.001, 0.0001]
Training model for 20 epochs.
Training 134608789 parameters in 34 parameter tensors.
-------------------------------------------------------------------
Build info:

                Built time: Sep 15 2017 07:42:54
                Last modified date: Thu Sep 14 22:33:54 2017
                Build type: Release
                Build target: GPU
                With 1bit-SGD: no
                With ASGD: yes
                Math lib: mkl
                CUDA version: 8.0.0
                CUDNN version: 6.0.21
                Build Branch: HEAD
                Build SHA1: 23878e5d1f73180d6564b6f907b14fe5f53513bb
                MPI distribution: Microsoft MPI
                MPI version: 7.0.12437.6
-------------------------------------------------------------------
PROGRESS: 0.00%
PROGRESS: 0.00%
PROGRESS: 0.00%
PROGRESS: 0.00%
Finished Epoch[1 of 20]: [Training] loss = nan * 20, metric = 78.45% * 20 84.281s (  0.2 samples/s);
PROGRESS: 0.00%
 : 
```

**Comment by kyoro1** [comment]
Just in case, here is the logs for `AlexNet` as base model. The training process seems to be correctly executed.

```sh
C:\git\ver2.2\CNTK\Examples\Image\Detection\FastRCNN (master)
(cntk2.2) λ python run_fast_rcnn.py
Selected GPU[0] Tesla K80 as the process wide default device.
Using base model:   AlexNet
lr_per_sample:      [0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.001, 0.001, 0.001, 0.001, 0.001, 0.001, 0.001, 0.001, 0.001, 0.001, 0.0001]
Training model for 20 epochs.
Training 57216469 parameters in 21 parameter tensors.
-------------------------------------------------------------------
Build info:

                Built time: Sep 15 2017 07:42:54
                Last modified date: Thu Sep 14 22:33:54 2017
                Build type: Release
                Build target: GPU
                With 1bit-SGD: no
                With ASGD: yes
                Math lib: mkl
                CUDA version: 8.0.0
                CUDNN version: 6.0.21
                Build Branch: HEAD
                Build SHA1: 23878e5d1f73180d6564b6f907b14fe5f53513bb
                MPI distribution: Microsoft MPI
                MPI version: 7.0.12437.6
-------------------------------------------------------------------
PROGRESS: 0.00%
Finished Epoch[1 of 20]: [Training] loss = 1.002074 * 20, metric = 18.60% * 20 10.859s (  1.8 samples/s);
PROGRESS: 0.00%
Finished Epoch[2 of 20]: [Training] loss = 0.711525 * 20, metric = 13.45% * 20 4.469s (  4.5 samples/s);
  :
```

**Comment by cha-zhang** [comment]
Are you running the example as is, or you are tuning parameters yourself?

**Comment by kyoro1** [comment]
Yes, and I asked this issue to @pkranen, and he advised us that we should change the learning parameters. Here is the result after changing as follows:

- For Fast R-CNN: `FastRCNN_config.py`
  - `__C.CNTK.LR_FACTOR = 1.0 # 10.0 is used for the Grocery example data. Start with 1.0 for other data sets.` The default parameter was `10.0`. After changing to `1.0`, the process was successful:)

So, can I send a pull-request for the script? (Indeed, Faster R-CNN is under the same situation.)

```sh
C:\git\ver2.2\CNTK\Examples\Image\Detection\FastRCNN (master)
(cntk2.2) λ python run_fast_rcnn.py
Selected GPU[0] Tesla K80 as the process wide default device.
Using base model:   VGG16
lr_per_sample:      [0.001, 0.001, 0.001, 0.001, 0.001, 0.001, 0.001, 0.001, 0.001, 0.001, 0.0001, 0.0001, 0.0001, 0.0001, 0.0001, 0.0001, 0.0001, 0.0001, 0.0001, 0.0001, 1e-05]
Training model for 3 epochs.
Training 134608789 parameters in 34 parameter tensors.
-------------------------------------------------------------------
Build info:

                Built time: Sep 15 2017 07:42:54
                Last modified date: Thu Sep 14 22:33:54 2017
                Build type: Release
                Build target: GPU
                With 1bit-SGD: no
                With ASGD: yes
                Math lib: mkl
                CUDA version: 8.0.0
                CUDNN version: 6.0.21
                Build Branch: HEAD
                Build SHA1: 23878e5d1f73180d6564b6f907b14fe5f53513bb
                MPI distribution: Microsoft MPI
                MPI version: 7.0.12437.6
-------------------------------------------------------------------
PROGRESS: 0.00%
PROGRESS: 0.00%
PROGRESS: 0.00%
Finished Epoch[1 of 3]: [Training] loss = 1.032923 * 20, metric = 18.92% * 20 52.696s (  0.4 samples/s);
PROGRESS: 0.00%
PROGRESS: 0.00%
Finished Epoch[2 of 3]: [Training] loss = 0.552913 * 20, metric = 11.60% * 20 18.656s (  1.1 samples/s);
PROGRESS: 0.00%
PROGRESS: 0.00%
Finished Epoch[3 of 3]: [Training] loss = 0.427450 * 20, metric = 8.77% * 20 18.687s (  1.1 samples/s);
creating eval model
0:01:37.124881
Evaluating Fast R-CNN model for 5 images.
Number of rois before non-maximum suppression: 64
Number of rois  after non-maximum suppression: 27
AP for         ketchup = 0.0000
AP for         joghurt = 1.0000
AP for          eggBox = 0.9500
AP for         tabasco = 0.0000
AP for         mustard = 0.0000
AP for          butter = 0.3333
AP for          tomato = 0.0000
AP for           water = 0.0000
AP for          pepper = 0.2000
AP for         avocado = 0.6667
AP for       champagne = 0.5000
AP for          gerkin = 1.0000
AP for           onion = 0.0000
AP for          orange = 0.0000
AP for     orangeJuice = 0.7500
AP for            milk = 0.3333
Mean AP = 0.3583
Plotting results from for 5 images.
```

**Comment by cha-zhang** [comment]
@kyoro1 Yes, please send us a PR. Thanks!
